pkg/registry/apis/provisioning/jobs/delete/worker.go

```diff
@@ -157,3 +160,22 @@ func (w *Worker) resolveResourcesToPaths(ctx context.Context, rw repository.Read
 
 	return resolvedPaths, nil
 }
+
+// deduplicatePaths removes duplicate file paths from the slice while preserving order
+func deduplicatePaths(paths []string) []string {
+	if len(paths) <= 1 {
+		return paths
+	}
+
+	seen := make(map[string]bool, len(paths))
```

Comment:
FYI (no need to change) – it's customary (from my experience / knowledge) in Go to use `map[T]struct{}` when you don't care about the values and you just want a unique index into something.

Just spreading some information! 😄